### PR TITLE
[consensus] replace ProposerInfo with Author

### PR DESCRIPTION
**Pull request by zekun000**
## Motivation

Replace the trait with concrete struct for better readability, we can still extend the struct to include more info in the future.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Yes.

## Test Plan

Existing tests.

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/libra/website, and link to your PR here.)

### CODE REVIEW

**Comment by andll** [comment]
This looks good to me. Quick q to @dmitri-perelman - is there any future reason you can think of to keep ProposerInfo as separate struct(with a single field being `Author`) vs just replacing it with Author?

**Comment by dmitri-perelman** [comment]
The original motivation to have a struct with ProposerInfo was to support things like VRF scores, in which an author can generate information that cannot be generated by anyone else.
We're not planning to use these ideas in the short term, hence stripping the unnecessary boilerplate and keeping plain `Author` field seems reasonable (although I'd probably prefer to first land the NIL block PRs to avoid unnecessary merge conflicts :))

File: consensus/src/chained_bft/consensus_types/proposal_info.rs [hunk 1]
```diff
@@ -10,35 +10,19 @@ use network::proto::Proposal as ProtoProposal;
...
-}
-
 /// ProposalInfo contains the required information for the proposer election protocol to make its
 /// choice (typically depends on round and proposer info).
 #[derive(Debug, Clone, PartialEq, Eq)]
-pub struct ProposalInfo<T, P> {
+pub struct ProposalInfo<T> {
     pub proposal: Block<T>,
-    pub proposer_info: P,
+    pub proposer_info: Author,
```
[the remark this answers is not in the document]
Comment (aching) [reply consensus/src/chained_bft/consensus_types/proposal_info.rs]: In future work (not this PR), let's figure out how to sign the proposals, votes, and timeout messages and put this type of validation to before these types of messages are passed to consensus.

[the remark this answers is not in the document]
Comment (aching) [reply consensus/src/chained_bft/consensus_types/proposal_info.rs]: Added Issue #479 to track this unrelated issue.

**Comment by zekun000** [comment]
Remove proposer_info from ProposalInfo, rename ProposalInfo to ProposalMsg to be consistent with TimeoutMsg, VoteMsg

File: consensus/src/chained_bft/chained_bft_smr.rs [hunk 2]
```diff
@@ -76,15 +73,15 @@ impl ChainedBftSMRConfig {
     }
 }
 
-/// ChainedBFTSmr is the one to generate the components (BLockStore, Proposer, etc.) and start the
+/// ChainedBFTSmr is the one to generate the components (BLockStoreroposer, etc.) and start the
```
Comment (aching) [inline consensus/src/chained_bft/chained_bft_smr.rs]:
```suggestion
/// ChainedBftSMR is the one to generate the components (BlockStore, Proposer, etc.) and start the
```

**Comment by zekun000** [comment]
Rebase

**Pull request merged**